Makefile

```diff
@@ -177,70 +193,61 @@ clean:
-		-w /go/src/$(PACKAGE_NAME) \
-		$(CALICO_BUILD) glide install -strip-vendor
+	$(DOCKER_RUN) $(CALICO_BUILD) go mod vendor
 
 ## Default the repos and versions but allow them to be overridden
 LIBCALICO_BRANCH?=$(shell git rev-parse --abbrev-ref HEAD)
 LIBCALICO_REPO?=github.com/projectcalico/libcalico-go
 LIBCALICO_VERSION?=$(shell git ls-remote git@github.com:projectcalico/libcalico-go $(LIBCALICO_BRANCH) 2>/dev/null | cut -f 1)
+LIBCALICO_OLD?=$(shell grep libcalico-go go.mod | cut -d' ' -f2)
 FELIX_BRANCH?=$(shell git rev-parse --abbrev-ref HEAD)
 FELIX_REPO?=github.com/projectcalico/felix
 FELIX_VERSION?=$(shell git ls-remote git@github.com:projectcalico/felix $(FELIX_BRANCH) 2>/dev/null | cut -f 1)
+FELIX_OLD?=$(shell grep felix go.mod | cut -d' ' -f2)
 CONFD_BRANCH?=$(shell git rev-parse --abbrev-ref HEAD)
 CONFD_REPO?=github.com/projectcalico/confd
 CONFD_VERSION?=$(shell git ls-remote git@github.com:projectcalico/confd $(CONFD_BRANCH) 2>/dev/null | cut -f 1)
+CONFD_OLD?=$(shell grep confd go.mod | cut -d' ' -f2)
 
-### Update pins in glide.yaml
 update-felix-confd-libcalico:
-	docker run --rm \
-        -v $(CURDIR):/go/src/$(PACKAGE_NAME):rw $$EXTRA_DOCKER_BIND \
-        -v $(HOME)/.glide:/home/user/.glide:rw \
-        -w /go/src/$(PACKAGE_NAME) \
-        -e LOCAL_USER_ID=$(LOCAL_USER_ID) \
-        $(CALICO_BUILD) sh -c '\
-          echo "Updating libcalico to $(LIBCALICO_VERSION) from $(LIBCALICO_REPO)"; \
-          echo "Updating felix to $(FELIX_VERSION) from $(FELIX_REPO)"; \
-          echo "Updating confd to $(CONFD_VERSION) from $(CONFD_REPO)"; \
-          export OLD_LIBCALICO_VER=$$(grep --after 50 libcalico glide.yaml |grep --max-count=1 --only-matching --perl-regexp "version:\s*\K[\.0-9a-z]+") ;\
-          export OLD_FELIX_VER=$$(grep --after 50 felix glide.yaml |grep --max-count=1 --only-matching --perl-regexp "version:\s*\K[\.0-9a-z]+") ;\
-          export OLD_CONFD_VER=$$(grep --after 50 confd glide.yaml |grep --max-count=1 --only-matching --perl-regexp "version:\s*\K[\.0-9a-z]+") ;\
-          echo "Old libcalico version: $$OLD_LIBCALICO_VER";\
-          echo "Old felix version: $$OLD_FELIX_VER";\
-          echo "Old confd version: $$OLD_CONFD_VER";\
-          if [ $(LIBCALICO_VERSION) != $$OLD_LIBCALICO_VER ] || [ $(FELIX_VERSION) != $$OLD_FELIX_VER ] || [ $(CONFD_VERSION) != $$OLD_CONFD_VER ]; then \
-            sed -i "s/$$OLD_LIBCALICO_VER/$(LIBCALICO_VERSION)/" glide.yaml && \
-            sed -i "s/$$OLD_FELIX_VER/$(FELIX_VERSION)/" glide.yaml && \
-            sed -i "s/$$OLD_CONFD_VER/$(CONFD_VERSION)/" glide.yaml && \
-            glide up --strip-vendor || glide up --strip-vendor; \
-          fi'
+ifneq ($(strip $(LIBCALICO_VERSION)),)
+ifneq ($(strip $(LIBCALICO_OLD)),)
+	@echo "Updating libcalico from $(LIBCALICO_OLD) to $(LIBCALICO_VERSION) using $(LIBCALICO_REPO)"
+	$(DOCKER_RUN) $(CALICO_BUILD) sed -i "/libcalico-go/d" go.mod go.sum && go get $(LIBCALICO_REPO)@$(LIBCALICO_VERSION)
```

Comment:
instead of `sed` the "go" way to do this would be

```
go mod edit -droprequire github.com/projectcalico/libcalico-go && go get...
```

Reply:
Fixed